test/sequential/test-inspector-esm.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+const common = require('../common');
+
+common.skipIfInspectorDisabled();
+
+const assert = require('assert');
+const fixtures = require('../common/fixtures');
+const { NodeInstance } = require('../common/inspector-helper.js');
+
+function checkListResponse(response) {
+  assert.strictEqual(1, response.length);
+  assert.ok(response[0]['devtoolsFrontendUrl']);
+  assert.ok(
+    /ws:\/\/127\.0\.0\.1:\d+\/[0-9A-Fa-f]{8}-/
+      .test(response[0]['webSocketDebuggerUrl']));
+}
+
+function checkVersion(response) {
+  assert.ok(response);
+  const expected = {
+    'Browser': `node.js/${process.version}`,
+    'Protocol-Version': '1.1',
+  };
+  assert.strictEqual(JSON.stringify(response),
+                     JSON.stringify(expected));
+}
+
+function checkBadPath(err) {
+  assert(err instanceof SyntaxError);
+  assert(/Unexpected token/.test(err.message), err.message);
+  assert(/WebSockets request was expected/.test(err.body), err.body);
+}
+
+function assertNoUrlsWhileConnected(response) {
+  assert.strictEqual(1, response.length);
+  assert.ok(!response[0].hasOwnProperty('devtoolsFrontendUrl'));
+  assert.ok(!response[0].hasOwnProperty('webSocketDebuggerUrl'));
+}
+
+function assertScopeValues({ result }, expected) {
+  const unmatched = new Set(Object.keys(expected));
+  for (const actual of result) {
+    const value = expected[actual['name']];
+    if (value) {
+      assert.strictEqual(value, actual['value']['value']);
+      unmatched.delete(actual['name']);
+    }
+  }
+  if (unmatched.size)
```

Comment:
Could you write this as `assert.strictEqual(unmatched.size, 0)`?

Reply:
Changed to comparison to an empty array.